### PR TITLE
Pimp the modern cmake examples to be modern

### PR DESCRIPTION
**Pull request by ClausKlein**
Prevent most cmakelint warnings.
Prevent to download gtest if found on build host.
Use modern boost version 1.71.0
Use ctest too.

Note: the test fails!

### CODE REVIEW

File: example_exec/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.13..3.16 FATAL_ERROR)
```
Comment (mathstuf) [inline example_exec/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.13...3.16 FATAL_ERROR)
```

File: libjsonutils/CMakeLists.txt [hunk 2]
```diff
@@ -1,19 +1,24 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.13..3.16 FATAL_ERROR)
```
Comment (mathstuf) [inline libjsonutils/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.13...3.16 FATAL_ERROR)
```

File: libjsonutils/cmake/FindRapidJSON.cmake [hunk 3]
```diff
@@ -1,46 +1,55 @@
...
 #
+# from http://rapidjson.org/
+#
 # This will define the following variables
 #
-#    RapidJSON_FOUND
-#    RapidJSON_INCLUDE_DIRS
+#    RAPIDJSON_FOUND
+#    RAPIDJSON_VERSION
+#    RAPIDJSON_INCLUDE_DIRS
```
Comment (mathstuf) [inline libjsonutils/cmake/FindRapidJSON.cmake]: Why was the case changed? These should probably stay `RapidJSON`.

File: libjsonutils/cmake/FindRapidJSON.cmake [hunk 4]
```diff
@@ -1,46 +1,55 @@
...
 #
 # and the following imported targets
 #
 #     RapidJSON::RapidJSON
 #
 # Author: Pablo Arias - pabloariasal@gmail.com
+#
+# clausklein$ cmakelint *.cmake
+# FindRapidJSON.cmake:0: Find modules should use uppercase names;
+#   consider using FindRAPIDJSON.cmake [convention/filename]
```
Comment (mathstuf) [inline libjsonutils/cmake/FindRapidJSON.cmake]: What convention is this? It sounds really weird to me.

File: libjsonutils/cmake/FindRapidJSON.cmake [hunk 5]
```diff
@@ -1,46 +1,55 @@
...
+# Total Errors: 1
 
-find_package(PkgConfig)
+find_package(PkgConfig REQUIRED)
```
Comment (mathstuf) [inline libjsonutils/cmake/FindRapidJSON.cmake]: I don't think this makes sense. You're failing the entire configure even for a non-required `RapidJSON` finding.

File: libjsonutils/cmake/FindRapidJSON.cmake [hunk 6]
```diff
@@ -1,46 +1,55 @@
...
     PATHS ${PC_RapidJSON_INCLUDE_DIRS}
     PATH_SUFFIXES rapidjson
 )
 
-set(RapidJSON_VERSION ${PC_RapidJSON_VERSION})
+set(RAPIDJSON_VERSION ${PC_RapidJSON_VERSION})
 
-mark_as_advanced(RapidJSON_FOUND RapidJSON_INCLUDE_DIR RapidJSON_VERSION)
+# NOTE: (cmakelint) All vars should be uppercase! CK
+mark_as_advanced(RAPIDJSON_FOUND RAPIDJSON_INCLUDE_DIR RAPIDJSON_VERSION)
```
Comment (mathstuf) [inline libjsonutils/cmake/FindRapidJSON.cmake]: Uhh, no. That hasn't been the case in a long time. Variables are better matched to the module name.

File: libjsonutils/cmake/FindRapidJSON.cmake [hunk 7]
```diff
@@ -1,46 +1,55 @@
...
+find_package_handle_standard_args(RAPIDJSON
+    REQUIRED_VARS RAPIDJSON_INCLUDE_DIR
+    VERSION_VAR RAPIDJSON_VERSION
 )
 
-if(RapidJSON_FOUND)
+if(RAPIDJSON_FOUND)
     #Set include dirs to parent, to enable includes like #include <rapidjson/document.h>
-    get_filename_component(RapidJSON_INCLUDE_DIRS ${RapidJSON_INCLUDE_DIR} DIRECTORY)
+    get_filename_component(RAPIDJSON_INCLUDE_DIRS ${RAPIDJSON_INCLUDE_DIR} DIRECTORY)
```
Comment (mathstuf) [inline libjsonutils/cmake/FindRapidJSON.cmake]: Better is to just do `find_path(NAMES rapidjson/rapidjson.h)` which will get you the parent directory automatically.

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in [hunk 8]
```diff
@@ -1,18 +1,18 @@
...
-#find_dependency(Boost 1.55 REQUIRED COMPONENTS regex)
-#find_dependency(RapidJSON 1.0 REQUIRED MODULE)
-find_package(Boost 1.55 REQUIRED COMPONENTS regex)
-find_package(RapidJSON 1.0 REQUIRED MODULE)
+get_filename_component(JSONUTILS_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
+#
+# NOTE: We had to use find_package because find_dependency does not support
+# COMPONENTS or MODULE until 3.15.0? CK
+#
+list(APPEND CMAKE_MODULE_PATH ${JSONUTILS_CMAKE_DIR})
```
Comment (mathstuf) [inline libjsonutils/cmake/JSONUtilsConfig.cmake.in:4]: I suggest prepending the path. You want to prefer your module over any other around right?

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in [hunk 9]
```diff
@@ -1,18 +1,18 @@
...
+# NOTE: to find FindRapidJSON.cmake
+find_package(RAPIDJSON @MIN_RAPIDJSON_VERSION@ REQUIRED MODULE)
```
Comment (mathstuf) [inline libjsonutils/cmake/JSONUtilsConfig.cmake.in]: This has to be `RapidJSON` to match the case of the find module. Also, the `REQUIRED` has poor semantics when finding `JSONUtils` as an optional package (or `QUIET` for that matter).

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in [hunk 10]
```diff
@@ -1,18 +1,18 @@
...
 list(REMOVE_AT CMAKE_MODULE_PATH -1)
```
Comment (mathstuf) [inline libjsonutils/cmake/JSONUtilsConfig.cmake.in:7]: `FindRapidJSON` may have modified `CMAKE_MODULE_PATH`. I suggest saving the old value, restoring it, then `unset()`'ing your temporary variable.

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in [hunk 11]
```diff
@@ -1,18 +1,18 @@
...
 
+include(CMakeFindDependencyMacro)
+find_dependency(boost_regex @MIN_BOOST_VERSION@ REQUIRED CONFIG)
```
Comment (mathstuf) [inline libjsonutils/cmake/JSONUtilsConfig.cmake.in]: Why are you passing `REQUIRED`? `find_dependency` handles that for you.

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in [hunk 12]
```diff
@@ -1,18 +1,18 @@
...
+
 if(NOT TARGET JSONUtils::JSONUtils)
-    include("${JSONUtils_CMAKE_DIR}/JSONUtilsTargets.cmake")
+    include("${JSONUTILS_CMAKE_DIR}/JSONUtilsTargets.cmake")
 endif()
 
-set(JSONUtils_LIBRARIES JSONUtils::JSONUtils)
+set(JSONUTILS_lIBRARIES JSONUtils::JSONUtils)
```
Comment (mathstuf) [inline libjsonutils/cmake/JSONUtilsConfig.cmake.in:15]: It is far better to match the case with the module name. This should also only be done if the module is found (basically, the dependencies exist).

File: libjsonutils/src/json_utils.cpp [hunk 13]
```diff
@@ -1,23 +1,24 @@
...
-    boost::optional<rapidjson::Document> loadFromUrl(const std::string& url)
+    boost::optional<rapidjson::Document> loadFromUrl(const std::string& /* url */)
     {
         //Download resource and extract message body
         //Return boost::none if something went wrong, e.g. status code >= 300
         rapidjson::Document doc;
         doc.Parse("{\"source\": \"url\"}");
 
-        return doc;
+        return std::move(doc);
```
Comment (mathstuf) [inline libjsonutils/src/json_utils.cpp:21]: These changes seem like they belong in a separate commit at least, probably a separate PR even.

File: libjsonutils/test/CMakeLists.txt [hunk 14]
```diff
@@ -1,25 +1,36 @@
...
-  RESULT_VARIABLE result
-  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/googletest-download )
-if(result)
-  message(FATAL_ERROR "Build step for googletest failed: ${result}")
-endif()
+# see /opt/local/share/cmake-3.16/Modules/FindGTest.cmake
+find_package(GTEST QUIET)
+
+if(NOT GTEST_FOUND)
+  # Download and unpack googletest at configure time
```
Comment (mathstuf) [inline libjsonutils/test/CMakeLists.txt:6]: I think `FetchContent` is the modern way to do this now. I suggest asking on the Slack channel for the recommended way these days.

**Review by mathstuf — CHANGES_REQUESTED** [review]

File: example_exec/CMakeLists.txt (hunk 1 above)
Comment (pabloariasal) [reply example_exec/CMakeLists.txt]: three `...` are required indeed

File: libjsonutils/CMakeLists.txt [hunk 15]
```diff
@@ -1,19 +1,24 @@
...
+
+set(MIN_RAPIDJSON_VERSION 1.1)
+find_package(RapidJSON ${MIN_RAPIDJSON_VERSION} REQUIRED MODULE)
 
 ##############################################
 # Create target and set properties
 
 add_library(jsonutils
     src/json_utils.cpp
+    src/file_utils.h
```
Comment (pabloariasal) [inline libjsonutils/CMakeLists.txt:21]: is this for IDEs that parse the sources, or why is this necessary?

File: libjsonutils/cmake/FindRapidJSON.cmake (hunk 3 above)
Comment (pabloariasal) [reply libjsonutils/cmake/FindRapidJSON.cmake]: I also consider this non-standard, I would stick to the convention used by `cmake` itself:

```
$ grep -m 1 Boost_FOUND /usr/share/cmake-3.16/Modules/FindBoost.cmake
  Boost_FOUND            - True if headers and requested libraries were found
```

File: libjsonutils/cmake/FindRapidJSON.cmake (hunk 4 above)
Comment (pabloariasal) [reply libjsonutils/cmake/FindRapidJSON.cmake]: I find it very strange that cmakelint flags this perfectly standard naming convention... Is [this](https://github.com/richq/cmake-lint) what you are using? maybe some configuration issue?

File: libjsonutils/cmake/FindRapidJSON.cmake (hunk 5 above)
Comment (pabloariasal) [reply libjsonutils/cmake/FindRapidJSON.cmake]: Agree with @mathstuf, PkgConfig is completely optional in this particular case

File: libjsonutils/cmake/FindRapidJSON.cmake (hunk 7 above)
Comment (pabloariasal) [reply libjsonutils/cmake/FindRapidJSON.cmake]: yes, we already did `find_path` above to find rapidjson's include directory, this gives you `/usr/linclude/rapidjson`, which is correct, the problem is that we can't do `#include <rapidjson/document.h>` in the code, but only `#include <document.h>`, which a little inconvenient as the reader doesn't have the library name when exporting. So here we set the actual `RapidJSON_INCLUDE_DIRS` to one level above: `/usr/include`

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in (hunk 11 above)
Comment (pabloariasal) [reply libjsonutils/cmake/JSONUtilsConfig.cmake.in]: True, `find_dependency` should handle this for you https://cmake.org/cmake/help/v3.15/module/CMakeFindDependencyMacro.html

File: libjsonutils/CMakeLists.txt [hunk 16]
```diff
@@ -1,19 +1,24 @@
...
 #Make sure that custom modules like FindRapidJSON are found
 list(INSERT CMAKE_MODULE_PATH 0 ${CMAKE_SOURCE_DIR}/cmake)
 
 ##############################################
 # Declare dependencies
-find_package(Boost 1.55 REQUIRED COMPONENTS regex)
-find_package(RapidJSON 1.0 REQUIRED MODULE)
+set(MIN_BOOST_VERSION 1.71)
+find_package(boost_headers ${MIN_BOOST_VERSION} REQUIRED CONFIG)
+find_package(boost_regex ${MIN_BOOST_VERSION} REQUIRED CONFIG)
```
Comment (pabloariasal) [inline libjsonutils/CMakeLists.txt:11]: what about this?
```suggestion
find_package(Boost ${MIN_BOOST_VERSION} REQUIRED CONFIG COMPONENTS headers regex)
```
I would like to still have an example of finding a single components in a modularized library via `COMPONENTS`

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in (hunk 9 above)
Comment (pabloariasal) [inline libjsonutils/cmake/JSONUtilsConfig.cmake.in]: question, does `find_dependency(RapidJSON @MIN_RAPIDJSON_VERSION@ MODULE)` fail on cmake-3.13? which cmake version are you using? I think we should bump min required cmake version to the one that allows us to (properly) make use of `find_dependency` here

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in [hunk 17]
```diff
@@ -1,18 +1,18 @@
...
-
-# NOTE Had to use find_package because find_dependency does not support COMPONENTS or MODULE until 3.8.0
-
-#find_dependency(Boost 1.55 REQUIRED COMPONENTS regex)
-#find_dependency(RapidJSON 1.0 REQUIRED MODULE)
-find_package(Boost 1.55 REQUIRED COMPONENTS regex)
-find_package(RapidJSON 1.0 REQUIRED MODULE)
+get_filename_component(JSONUTILS_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
+#
+# NOTE: We had to use find_package because find_dependency does not support
```
Comment (pabloariasal) [inline libjsonutils/cmake/JSONUtilsConfig.cmake.in]: see comment below, maybe we should bump the version to 3.15 and use `find_dependency` here correctly, what do you say?

File: libjsonutils/src/json_utils.cpp (hunk 13 above)
Comment (pabloariasal) [inline libjsonutils/src/json_utils.cpp:21]: Never `std::move()` a local variable http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f48-dont-return-stdmovelocal
It may inhibit RVO
```suggestion
        return doc;
```

File: libjsonutils/test/CMakeLists.txt (hunk 14 above)
Comment (pabloariasal) [reply libjsonutils/test/CMakeLists.txt:6]: yes, this might be already quite old, these days I don't even now if gtest must be downloaded and compiled like this, I get mine from conan today. Even so, I would consider this a topic of another pull request

File: libjsonutils/test/CMakeLists.txt [hunk 18]
```diff
@@ -1,25 +1,36 @@
...
-if(result)
-  message(FATAL_ERROR "CMake step for googletest failed: ${result}")
-endif()
-execute_process(COMMAND ${CMAKE_COMMAND} --build .
-  RESULT_VARIABLE result
-  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/googletest-download )
-if(result)
-  message(FATAL_ERROR "Build step for googletest failed: ${result}")
-endif()
+# see /opt/local/share/cmake-3.16/Modules/FindGTest.cmake
```
Comment (pabloariasal) [inline libjsonutils/test/CMakeLists.txt:1]: this is system dependent, I would remove this comment
```suggestion
```

File: libjsonutils/test/src/main.cpp [hunk 19]
```diff
@@ -1,6 +1,7 @@
-#include <gtest/gtest.h>
 #include <jsonutils/json_utils.h>
 
+#include <gtest/gtest.h>
```
Comment (pabloariasal) [inline libjsonutils/test/src/main.cpp]: I would stick to the [Google's C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) and include third party headers before your own

File: libjsonutils/src/json_utils.cpp [hunk 20]
```diff
@@ -31,19 +32,19 @@ namespace jsonutils
         rapidjson::Document doc;
         doc.Parse("{\"source\": \"file\"}");
 
-        return doc;
+        return std::move(doc);
```
Comment (pabloariasal) [inline libjsonutils/src/json_utils.cpp:36]:
```suggestion
       // force-move into the optional, as some compilers don't do this automatically
        return std::move(doc);
```

File: libjsonutils/test/CMakeLists.txt [hunk 21]
```diff
@@ -1,25 +1,36 @@
...
-                 EXCLUDE_FROM_ALL)
+  execute_process(COMMAND ${CMAKE_COMMAND} --build .
+    RESULT_VARIABLE result
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/googletest-download)
+  if(result)
+    message(FATAL_ERROR "Build step for googletest failed: ${result}")
+  endif()
+
+  # Add googletest directly to our build. This defines
+  # the gtest and gtest_main targets.
```
Comment (pabloariasal) [inline libjsonutils/test/CMakeLists.txt:25]: are you sure the names of these targets are correct? below you use `GTest::Main`

File: libjsonutils/src/json_utils.cpp [hunk 22]
```diff
@@ -1,23 +1,24 @@
+#include "file_utils.h"
+
 #include "jsonutils/json_utils.h"
 
 #include <boost/regex.hpp>
 #include <iostream>
 
-#include <file_utils.h>
```
Comment (pabloariasal) [inline libjsonutils/src/json_utils.cpp:6]: see comment below regarding order of includes

File: example_exec/CMakeLists.txt (hunk 1 above)
Comment (pabloariasal) [inline example_exec/CMakeLists.txt]: Do you know which cmake version introduces `MODULE` and `COMPONENTS` for `find_dependency`? I think this should be our minimum required version. Also, any reason to restrict it to a max?

**Review by pabloariasal — CHANGES_REQUESTED** [review]
Thanks a lot for this pull request! and @mathstuf for helping out in the review. Most of the changes are very reasonable, I have two main open points:

1) I would leave out the renaming of all variables to upper case, specially the ones set in FindModules. As I commented below, this is not standard practice and we should do what cmake does in their FindModules: `Boost_FOUND` and not `BOOST_FOUND`. Maybe we can disable this linter warning and provide a cmakelintrc? But then I would make this a separate pull request

2) `find_package` in `JsonUtilsConfig.cmake` should be avoided, in favor of `find_dependency`, so I think we should bump to a version high enough that allows this

File: libjsonutils/cmake/FindRapidJSON.cmake (hunk 4 above)
Comment (ClausKlein) [reply libjsonutils/cmake/FindRapidJSON.cmake]: https://pypi.org/project/cmakelint/

File: libjsonutils/src/json_utils.cpp (hunk 13 above)
Comment (ClausKlein) [reply libjsonutils/src/json_utils.cpp:21]: Yes, but without this, it does't compile (copy constructor is deleted in boost now).
I changed it to continue testing with cmake.

File: libjsonutils/test/CMakeLists.txt (hunk 14 above)
Comment (ClausKlein) [reply libjsonutils/test/CMakeLists.txt:6]: I have planed a second patch using FetchContent

File: libjsonutils/CMakeLists.txt (hunk 15 above)
Comment (ClausKlein) [reply libjsonutils/CMakeLists.txt:21]: Yes, than it is show in xcode and used while search in project

File: libjsonutils/test/src/main.cpp (hunk 19 above)
Comment (ClausKlein) [reply libjsonutils/test/src/main.cpp]: Maybe good for Google, but I prefer to check the header to be compilable.
see https://llvm.org/docs/CodingStandards.html#include-style

File: example_exec/CMakeLists.txt (hunk 1 above)
Comment (ClausKlein) [reply example_exec/CMakeLists.txt]: IMHO: Debian stable include only cmake version 3.13. 
This is he only reason for me not to write 3.14!

File: libjsonutils/src/json_utils.cpp (hunk 13 above)
Comment (pabloariasal) [reply libjsonutils/src/json_utils.cpp:21]: which version of boost, rapidjson and compiler are you using? this compiles just fine in my system

File: example_exec/CMakeLists.txt (hunk 1 above)
Comment (ClausKlein) [reply example_exec/CMakeLists.txt]: changed

File: libjsonutils/CMakeLists.txt (hunk 2 above)
Comment (ClausKlein) [reply libjsonutils/CMakeLists.txt]: changed

File: libjsonutils/src/json_utils.cpp (hunk 13 above)
Comment (pabloariasal) [reply libjsonutils/src/json_utils.cpp:21]: alright I was able to find the problem. This has nothing to do with `boost::optional` (which is both a copiable and movable type), but rather `rapidjson::Document`, which is a move only type. Modern compilers like mine (g++9.2.1) are smart enough to move the document into the `boost::optional`, yours doesn't and tries to copy it into the optional.

Comment (pabloariasal) [reply libjsonutils/src/json_utils.cpp:21]:
```suggestion
       // force-move into the optional, as some compilers don't do this automatically
        return std::move(doc);
```

File: libjsonutils/cmake/FindRapidJSON.cmake (hunk 3 above)
Comment (ClausKlein) [reply libjsonutils/cmake/FindRapidJSON.cmake]: Claus-MBP:example_exec clausklein$ egrep -w '[A-Z]+_FOUND' /opt/local/share/cmake-3.16/Modules/Find*.cmake -l | wc
    113     113    6014
Claus-MBP:example_exec clausklein$ egrep -w '\w+_FOUND' /opt/local/share/cmake-3.16/Modules/Find*.cmake -l | wc
    144     144    7688
Claus-MBP:example_exec clausklein$

File: libjsonutils/test/src/main.cpp (hunk 19 above)
Comment (pabloariasal) [reply libjsonutils/test/src/main.cpp]: mm ok, but then we need to be consistent everywhere, maybe as part of a second pull request that modifies just the order of includes?

File: libjsonutils/cmake/FindRapidJSON.cmake (hunk 3 above)
Comment (ClausKlein) [reply libjsonutils/cmake/FindRapidJSON.cmake]: Note: the FindBoost module is deprecated! new versions install cmake config files.

Comment (ClausKlein) [reply libjsonutils/cmake/FindRapidJSON.cmake]: But I revert to CamelCase

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in (hunk 10 above)
Comment (ClausKlein) [reply libjsonutils/cmake/JSONUtilsConfig.cmake.in:7]: OK

File: libjsonutils/cmake/JSONUtilsConfig.cmake.in (hunk 12 above)
Comment (ClausKlein) [reply libjsonutils/cmake/JSONUtilsConfig.cmake.in:15]: OK

File: libjsonutils/test/CMakeLists.txt (hunk 21 above)
Comment (ClausKlein) [reply libjsonutils/test/CMakeLists.txt:25]: fixed

File: example_exec/CMakeLists.txt (hunk 1 above)
Comment (pabloariasal) [reply example_exec/CMakeLists.txt]: mm alright, I see that Ubuntu 18 ships version 3.10, so maybe we can leave it at 3.13 or even 3.10, and keep using `find_package` in the config.cmake

File: libjsonutils/src/json_utils.cpp (hunk 13 above)
Comment (ClausKlein) [reply libjsonutils/src/json_utils.cpp:21]: done
Claus-MBP:example_exec clausklein$ clang --version
clang version 9.0.0 (trunk 365158)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /opt/local/libexec/llvm-devel/bin

File: libjsonutils/test/src/main.cpp (hunk 19 above)
Comment (ClausKlein) [reply libjsonutils/test/src/main.cpp]: ok

File: libjsonutils/cmake/FindRapidJSON.cmake (hunk 5 above)
Comment (ClausKlein) [reply libjsonutils/cmake/FindRapidJSON.cmake]: removed

File: libjsonutils/src/json_utils.cpp (hunk 22 above)
Comment (ClausKlein) [reply libjsonutils/src/json_utils.cpp:6]: done

**Review by pabloariasal — APPROVED** [review]

**Comment by pabloariasal** [comment]
Vielen Dank @ClausKlein

**Comment by ClausKlein** [comment]
But there is still a problem with the test:

cd ../.build-cookbook-Debug && ctest -C Debug --rerun-failed --output-on-failure .
Test project /Users/clausklein/cmake/modern-cmake-sample/.build-cookbook-Debug
    Start 1: json_utils_test
1/1 Test #1: json_utils_test ..................***Exception: SegFault  0.32 sec
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from JsonUtilsTest


0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.33 sec

The following tests FAILED:
	  1 - json_utils_test (SEGFAULT)
Errors while running CTest


„Es ist nicht genug, zu wissen, man muß auch anwenden; 
     es ist nicht genug, zu wollen, man muß auch tun.“ 
Johann Wolfgang von Goethe <http://www.zitate-online.de/autor/goethe-johann-wolfgang-von/> (Werk: Wilhelm Meisters Wanderjahre)

**Comment by pabloariasal** [comment]
In my system they run just fine
```
  ~/W/modern-cmake-sample/l/build   master *1 ?1 ❯ ctest
Test project /home/pablo/Workspace/modern-cmake-sample/libjsonutils/build
    Start 1: json_utils_test
1/1 Test #1: json_utils_test ..................   Passed    0.01 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.02 sec
```

**Comment by pabloariasal** [comment]
What does GDB say? backtrace?

File: example_exec/CMakeLists.txt (hunk 1 above)
Comment (mathstuf) [reply example_exec/CMakeLists.txt]: The second version number is not a max supported version, but a max version awareness. It is "we're aware of this version, if you know about it, set policies to `NEW` that *max* knows about". https://cmake.org/cmake/help/latest/command/cmake_policy.html?highlight=cmake_policy#setting-policies-by-cmake-version

Comment (pabloariasal) [reply example_exec/CMakeLists.txt]: Alright I see thank you for the clarification :)
